File: Algoritmo(C)/Alg_S6_4.c
#include <stdio.h>
#include <ctype.h>
int main(){
    //variables
    float altura, pideal;
    char sexo;
    //input
    printf("Digite sua altura: ");
    scanf("%f",&altura);
    gets(stdin); //corrigir bug
    printf("Qual o seu sexo?[M/F] ");
    scanf("%c",&sexo);
    //process
    if (toupper(sexo) == 'M'){
        pideal = ((altura * 72.7) -58);
        printf("Seu peso ideal e: %.2f",pideal);
    }else if(toupper(sexo) == 'F'){
        pideal = ((altura * 62.1) -44.7);
        printf("Seu peso ideal e: %.2f",pideal);
    }else{
        printf("Sexo invalido.");
    }
}

File: Algoritmo(Python)/Alg_S6_4.py
#input
altura = float(input("Digite sua altura: "))
sexo = input("Qual é o seu sexo?[M/F] ")
#process
if(sexo.upper() == "M"):
    pideal = ((altura * 72.7) - 58)
elif(sexo.upper() == "F"):
    pideal = ((altura * 62.1) - 44.7)
else:
    pideal = 0
    print("Sexo inválido")
#output
print("O seu peso ideal é: {0:.2f}".format(pideal))